packages/wonder-blocks-dropdown/components/action-menu-opener-core.js

```diff
@@ -0,0 +1,183 @@
+// @flow
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+import PropTypes from "prop-types";
+
+import {LabelLarge} from "@khanacademy/wonder-blocks-typography";
+import Color, {SemanticColor, mix} from "@khanacademy/wonder-blocks-color";
+import {addStyle} from "@khanacademy/wonder-blocks-core";
+import Icon, {icons} from "@khanacademy/wonder-blocks-icon";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+
+import type {ClickableHandlers} from "@khanacademy/wonder-blocks-core";
+import type {SharedProps} from "./action-menu-opener.js";
+
+type Props = {|
+    ...SharedProps,
+    ...ClickableHandlers,
+    hovered: boolean,
+    focused: boolean,
+    pressed: boolean,
+|};
+
+const StyledButton = addStyle("button");
+
+export default class ActionMenuOpenerCore extends React.Component<Props> {
+    static contextTypes = {router: PropTypes.any};
+
+    render() {
+        const {
+            children,
+            disabled: disabledProp,
+            focused,
+            hovered,
+            pressed,
+            "aria-label": ariaLabel,
+            ...handlers
+        } = this.props;
+
+        const buttonColor = SemanticColor.controlDefault;
+        const buttonStyles = _generateStyles(buttonColor);
+        const disabled = disabledProp;
+
+        const defaultStyle = [
+            sharedStyles.shared,
+            disabled && sharedStyles.disabled,
+            buttonStyles.default,
+            disabled && buttonStyles.disabled,
+            !disabled &&
+                (pressed
+                    ? buttonStyles.active
+                    : (hovered || focused) && buttonStyles.focus),
+        ];
+
+        const commonProps = {
+            "aria-disabled": disabled ? "true" : undefined,
+            "aria-label": ariaLabel,
+            "data-test-id": "foo-bar",
+            role: "button",
+            style: [defaultStyle],
+            ...handlers,
+        };
+
+        const label = (
+            <LabelLarge style={sharedStyles.text}>
+                {children}
+                <Icon
+                    size="small"
+                    color="currentColor"
+                    icon={icons.caretDown}
+                    style={sharedStyles.icon}
+                />
+            </LabelLarge>
+        );
+
+        const contents = <React.Fragment>{label}</React.Fragment>;
+
+        return (
+            <StyledButton type="button" {...commonProps} disabled={disabled}>
+                {contents}
+            </StyledButton>
+        );
+    }
+}
+
+const sharedStyles = StyleSheet.create({
+    shared: {
+        position: "relative",
+        display: "inline-flex",
+        alignItems: "center",
+        justifyContent: "center",
+        height: 40,
+        paddingTop: 0,
+        paddingBottom: 0,
+        paddingLeft: 16,
+        paddingRight: 16,
+        border: "none",
+        borderRadius: 4,
```

Comment:
At least a few of these numbers could be Spacing constants!

Reply:
Will do.